### PR TITLE
docs(release.md): update steps

### PR DESCRIPTION
**Pull request by maxgio92**
Signed-off-by: maxgio92 <massimiliano.giovagnoli.1992@gmail.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This PR updates the release documentation, by:
- updating the Github release template and adding the changelog
- documenting the archiving of the website snapshot

**Special notes for your reviewer**:

The Github release description should contain also the changelog.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by leogr — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -111,3 +115,4 @@ Announce the new release to the world!
 - Publish a blog on [Falco website](https://github.com/falcosecurity/falco-website) ([example](https://github.com/falcosecurity/falco-website/blob/master/content/en/blog/falco-0-28-1.md))
 - Send an announcement to cncf-falco-dev@lists.cncf.io (plain text, please)
 - Let folks in the slack #falco channel know about a new release came out
+- If and only if, this release introduces a **new minor version**, [archive a snapshot of the Falco website](https://github.com/falcosecurity/falco-website/blob/master/release.md#documentation-versioning)
```
Comment (leodido) [inline RELEASE.md:118]:
```suggestion
- IFF the on going release introduces a **new minor version**, [archive a snapshot of the Falco website](https://github.com/falcosecurity/falco-website/blob/master/release.md#documentation-versioning)
```

File: RELEASE.md [hunk 2]
```diff
@@ -77,6 +77,10 @@ Now assume `x.y.z` is the new version.
     | `docker pull docker.io/falcosecurity/falco-driver-loader:x.y.z`             |
     | `docker pull docker.io/falcosecurity/falco-no-driver:x.y.z`                 |
 
+    <changelog>
+
+    <!-- Substitute changelog with the one generated by [rn2md](https://github.com/leodido/rn2md) -->
```
Comment (leodido) [inline RELEASE.md:82]:
```suggestion
    <!-- Substitute <changelog> with the one generated by [rn2md](https://github.com/leodido/rn2md) -->
```

**Review by leodido — COMMENTED** [review]
Left some nitpicks :)

**Comment by maxgio92** [comment]
Thanks @leodido and @leogr!

**Review by leogr — APPROVED** [review]

**Review by leodido — APPROVED** [review]

**Pull request merged**